mono/mini/mini-windows.c

```diff
@@ -71,6 +75,201 @@ typedef struct {
 	handler handler;
 } HandlerItem;
 
+#if _WIN64
+typedef void MONO_SIG_HANDLER_SIGNATURE ((*MonoW32ExceptionHandler));
+void win32_seh_init(void);
+void win32_seh_cleanup(void);
+void win32_seh_set_handler(int type, MonoW32ExceptionHandler handler);
+
+static void (*restore_stack) (void);
+static MonoW32ExceptionHandler fpe_handler;
+static MonoW32ExceptionHandler ill_handler;
+static MonoW32ExceptionHandler segv_handler;
+
+LPTOP_LEVEL_EXCEPTION_FILTER mono_old_win_toplevel_exception_filter;
+void *mono_win_vectored_exception_handle;
+
+#define W32_SEH_HANDLE_EX(_ex) \
+	if (_ex##_handler) _ex##_handler(er->ExceptionCode, &info, ctx)
+
+static LONG CALLBACK seh_unhandled_exception_filter(EXCEPTION_POINTERS* ep)
+{
+#ifndef MONO_CROSS_COMPILE
+	if (mono_old_win_toplevel_exception_filter) {
+		return (*mono_old_win_toplevel_exception_filter)(ep);
+	}
+#endif
+
+	if (mono_dump_start ())
+		mono_handle_native_crash (mono_get_signame (SIGSEGV), NULL, NULL);
+
+	return EXCEPTION_CONTINUE_SEARCH;
+}
+
+#if HAVE_API_SUPPORT_WIN32_RESET_STKOFLW && !TARGET_ARM64
```

Comment:
Let's create a bug so that we don't forget that ARM64 will be missing this impl.

Reply:
https://jira.unity3d.com/browse/UUM-20941

Reply:
Can you do a positive check for `TARGET_AMD64` rather than a negative check of `!TARGET_ARM64`?

Not blocking, but there is Win32 code path for this as well. If we are extracting these to this windows specific file, it would be nice to consolidate them all.